server/src/test/java/org/elasticsearch/index/shard/PrimaryReplicaSyncerTests.java

```diff
@@ -102,6 +123,54 @@ public void testSyncerSendsOffCorrectDocuments() throws Exception {
         closeShards(shard);
     }
 
+    public void testSyncerSendsMaxSeqNo() throws Exception {
```

Comment:
what does this test add on top of the check in https://github.com/elastic/elasticsearch/pull/30176/files#diff-4361011dc54c3c4a4545d2fd1d009529R103 ?